components/ViewHabits.tsx

```diff
@@ -35,35 +53,113 @@ const styles = {
     maxHeight: '16vh',
     maxWidth: '100vw',
   }),
+
   habitsContainer: css({
-    maxWidth: '500px',
+    maxWidth: '700px',
     margin: '0 auto',
   }),
-  addHabitButton: css({
-    borderRadius: '7px',
-    position: 'absolute',
-    padding: '0',
-    height: '48px',
-    width: '48px',
-    top: '65px',
-    right: '8px',
-    transform: 'translateY(-50%)',
+  weeklyContainer: css({
+    display: 'flex',
+    flexDirection: 'column',
+    border: '2.5px solid #fff',
+
+    tr: {
+      display: 'flex',
+      flexDirection: 'row',
+
+      justifySelf: 'space-between',
+      alignSelf: 'space-between',
+    },
+
+    tbody: {
+      td: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        flex: '1',
+        border: '0.5px solid #222',
+        padding: '0px 2px',
+        minHeight: '50px',
+        minWidth: '50px',
+        verticalAlign: 'middle',
+      },
+    },
+
+    th: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      flex: '1',
+      border: '0.5px solid #222',
+      backgroundColor: '#0066ff',
+      padding: '0px 2px',
+      minHeight: '50px',
+      minWidth: '50px',
+      verticalAlign: 'middle',
+    },
+
+    '& :first-of-type': {
+      flexBasis: '50%',
+    },
   }),
 };
 
+type TimeView = 'daily' | 'weekly';
+
 const ViewHabits = () => {
   const { t } = useTranslation(['common', 'add-habit']);
   const [habitsData, setHabitsData] = useState<GoalWithHabitHistory[]>([]);
+  const [timeView, setTimeView] = useState<TimeView>('daily');
+  const [dates, setDates] = useState<Date[]>([]);
```

Comment:
[curious] should this be an inferred/derived piece of data from habitData? I see that there's a `useEffect`, but it's the only time `setDates` is being used, and looks like a derivation rather than side-effect

Reply:
I'm not sure why I made that `useEffect()` like that... Changed `dates` to a `const` now.